### PR TITLE
Add tracing to smoke test environment

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

Added support to validate traces during execution of Smoke test. This will spin up
a Tempo instance where Agent will send traces. The validation and emition of traces
is made using [Vulture](https://github.com/grafana/tempo/blob/main/cmd/tempo-vulture/main.go).

#### PR Checklist

- [ ] CHANGELOG updated
- [x] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by mattdurham — APPROVED** [review]
LGTM

File: example/k3d/lib/tempo/main.libsonnet [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        '-mem-ballast-size-mbs=' + $._config.ballast_size_mbs,
+      ]) +
+      container.withVolumeMounts([
+        volumeMount.new(tempo_config_volume, '/conf'),
+        volumeMount.new(tempo_data_volume, '/var/tempo'),
+      ]) +
+      k.util.resourcesRequests('3', '3Gi') +
+      k.util.resourcesLimits('5', '5Gi'),
+
+    query_container::
```
Comment (mapno) [inline example/k3d/lib/tempo/main.libsonnet:80]: No need to deploy tempo-query

**Review by mapno — COMMENTED** [review]
Overall, it LGTM. But, couldn't we import the jsonnet/yamls from Tempo instead? This is something that will need to be maintained otherwise.

**Comment by marctc** [comment]
> But, couldn't we import the jsonnet/yamls from Tempo instead? This is something that will need to be maintained otherwise.

Good point, that would be lovely. @mattdurham @rfratto do you know if it's possible?

**Comment by mattdurham** [comment]
Your knowledge of jsonnet exceeds mine at this point =) I think you can import with github URLs but haven't tried it.

**Comment by rlankfo** [comment]
> > But, couldn't we import the jsonnet/yamls from Tempo instead? This is something that will need to be maintained otherwise.
> 
> Good point, that would be lovely. @mattdurham @rfratto do you know if it's possible?

You should be able to import dependencies from tempo by using `jb install` if I'm not mistaken. I believe the format is `jb install <repo.git>/<folder>@<branch>`.  

Here's an example I found in my history when adding agent yaml files to smoke:
```
jb install https://github.com/grafana/agent.git/production/tanka/grafana-agent@main
```

This imports (and vendors I believe) the `/production/tanka/grafana-agent` folder from `main`.

**Comment by marctc** [comment]
> > > But, couldn't we import the jsonnet/yamls from Tempo instead? This is something that will need to be maintained otherwise.
> > 
> > 
> > Good point, that would be lovely. @mattdurham @rfratto do you know if it's possible?
> 
> You should be able to import dependencies from tempo by using `jb install` if I'm not mistaken. I believe the format is `jb install <repo.git>/<folder>@<branch>`.
> 
> Here's an example I found in my history when adding agent jsonnet files to smoke:
> 
> ```
> jb install https://github.com/grafana/agent.git/production/tanka/grafana-agent@main
> ```
> 
> This imports (and vendors I believe) the `/production/tanka/grafana-agent` folder from `main`.

Thanks, I did this and I think i makes more sense. @mapno @rlankfo could you take another look?

File: example/k3d/smoke/main.jsonnet [hunk 2]
```diff
@@ -1,7 +1,10 @@
 local monitoring = import './monitoring/main.jsonnet';
 local cortex = import 'cortex/main.libsonnet';
+local tempo = import 'github.com/grafana/tempo/operations/jsonnet/single-binary/tempo.libsonnet';
 local avalanche = import 'grafana-agent/smoke/avalanche/main.libsonnet';
 local crow = import 'grafana-agent/smoke/crow/main.libsonnet';
+local vulture = import 'github.com/grafana/tempo/operations/jsonnet/microservices/vulture.libsonnet';
+local tempo = import 'github.com/grafana/tempo/operations/jsonnet/single-binary/tempo.libsonnet';
```
Comment (mapno) [inline example/k3d/smoke/main.jsonnet:7]: This line is duplicated in a few lines above

**Review by mapno — APPROVED** [review]
LGTM

**Review by tpaschalis — APPROVED** [review]

**Pull request merged**